hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/block/transformer/FileCreateTransformer.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2025 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.importer.downloader.block.transformer;
+
+import com.hedera.hapi.block.stream.output.protoc.StateIdentifier;
+import com.hedera.mirror.common.domain.transaction.BlockItem;
+import com.hedera.mirror.common.domain.transaction.TransactionType;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import com.hederahashgraph.api.proto.java.TransactionRecord;
+import jakarta.inject.Named;
+
+@Named
+final class FileCreateTransformer extends AbstractBlockItemTransformer {
+
+    @Override
+    protected void updateTransactionRecord(
+            BlockItem blockItem, TransactionRecord.Builder transactionRecordBuilder, TransactionBody transactionBody) {
+        for (var stateChange : blockItem.stateChanges()) {
+            for (var change : stateChange.getStateChangesList()) {
+                if (change.getStateId() == StateIdentifier.STATE_ID_FILES.getNumber()) {
+                    if (change.hasMapUpdate()) {
+                        var mapUpdate = change.getMapUpdate();
+                        if (mapUpdate.hasKey()) {
+                            var key = mapUpdate.getKey();
+                            if (key.hasFileIdKey()) {
+                                var fileId = key.getFileIdKey();
+                                transactionRecordBuilder.getReceiptBuilder().setFileID(fileId);
+                                return;
+                            }
+                        }
+                    }
+                }
```

Comment:
prefer less nested levels. for `MapUpdate.key`, `hasKey()` is unnecessary, because `getKey()` returns the default `MapChangeKey` instance, and `key.hasFileIdKey()` will be false.

```suggestion
                if (change.getStateId() == StateIdentifier.STATE_ID_FILES.getNumber() && change.hasMapUpdate()) {
                    var key = change.getMapUpdate().getKey();
                    if (key.hasFileIdKey()) {
                        transactionRecordBuilder.getReceiptBuilder().setFileID(key.getFileIdKey);
                        return;
                    }
                }
```

Reply:
Fixed